documentation/modules/auxiliary/scanner/scada/modbus_banner_grabbing.md

```diff
@@ -0,0 +1,68 @@
+## Vulnerable Application
+
+  Banner grabbing for Modbus using Function Code 43 (Read Device Identification). Modbus is a data communications protocol originally published by Modicon (now Schneider Electric) in 1979 for use with its programmable logic controllers (PLCs).
+  
+  https://en.wikipedia.org/wiki/Modbus#Available_function/command_codes
+
+## Verification Steps
+
+  1. Do: ```use auxiliary/scanner/scada/modbus_banner_grabbing```
+  2. Do: ```set RHOST=IP``` where IP is the IP address of the target.
+  2. Do: ```set UNIT_ID=ID``` where ID is a number between 1 and 254. This is optional, default Unite Identifier is set to ```0```.
+  3. Do: ```run```
+
+## Options
+
+```
+msf5 auxiliary(scanner/scada/modbus_banner_grabbing) > notes 
+
+Notes
+=====
+
+ Time                     Host            Service  Port  Protocol  Type                Data
+ ----                     ----            -------  ----  --------  ----                ----
+ 2020-07-06 13:25:50 UTC  192.168.1.1     modbus   502   tcp       modbus.vendorname   "Schneider Electric"
+ 2020-07-06 13:25:50 UTC  192.168.1.1     modbus   502   tcp       modbus.productcode  "BMX NOE 0100"
+ 2020-07-06 13:25:50 UTC  192.168.1.1     modbus   502   tcp       modbus.revision     "V3.10"
+ ```
```

Comment:
This section is used to describe module options rather than module output. This data should be formatted for markdown and consistency. The `./tools/dev/msftidy_docs.rb` utility can be used to validate a module documentation file. It will flag style violations.

* [Documentation](https://github.com/rapid7/metasploit-framework/wiki/Writing-Module-Documentation)
* [Template](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/module_doc_template.md)
* [Examples](https://github.com/rapid7/metasploit-framework/tree/master/documentation/modules)

Reply:
> This section is used to describe module options rather than module output. This data should be formatted for markdown and consistency. The `./tools/dev/msftidy_docs.rb` utility can be used to validate a module documentation file. It will flag style violations.
> 
> * [Documentation](https://github.com/rapid7/metasploit-framework/wiki/Writing-Module-Documentation)
> * [Template](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/module_doc_template.md)
> * [Examples](https://github.com/rapid7/metasploit-framework/tree/master/documentation/modules)

Hi, thank you for all the recommendations and the code refactoring :)

Reply:
Hi @bcoles I think the module is ready now :D